local-interchaintest/src/tests/remote_chain_splitter/remote_chain_splitter.rs

```diff
@@ -0,0 +1,307 @@
+use std::{
+    collections::BTreeMap,
+    time::{Duration, Instant},
+};
+use valence_covenant_single_party_pol::msg::DEFAULT_TIMEOUT;
+
+use log::info;
+
+struct ContractInfo {
+    code_id: u64,
+    contract_addr: String,
+}
+
+/// How long we will wait until the ICA must have been created, or timed out
+const ICA_TIMEOUT: Duration = Duration::from_secs(30);
+
+macro_rules! with_poll_timeout {
+    ($body:block) => {
+        let start = Instant::now();
+
+        while Instant::now().duration_since(start) < ICA_TIMEOUT {
+            $body
+        }
+    };
+}
+
+/// Tests that the remote chain splitter correctly handles
+/// - Normal case
+/// - Timeouts (does not advance to ica_created)
+/// - Timeout recovery (advances after on 2nd tick after a packet timeout)
+pub fn test_remote_chain_splitter(test_ctx: &mut TestContext) -> Result<(), LocalError> {
+    info!("Starting remote chain splitter tests...");
+
+    upload_contracts(test_ctx)?;
+
+    // See above: 3 tests, each with a different splitter
+    let splitters = (
+        make_remote_chain_splitter(test_ctx)?,
+        make_remote_chain_splitter(test_ctx)?,
+        make_remote_chain_splitter(test_ctx)?,
+    );
+    fund_remote_chain_splitter(test_ctx, &splitters.0)?;
+    fund_remote_chain_splitter(test_ctx, &splitters.1)?;
+    fund_remote_chain_splitter(test_ctx, &splitters.2)?;
+
+    // Separate tests: ensure that the splitter can handle the happy case
+    // and a timeout (independently)
+    test_remote_chain_splitter_ok(test_ctx, &splitters.0)?;
+    test_remote_chain_splitter_timeout(test_ctx, &splitters.1)?;
+
+    // Combined test: runs test_timeout and test_ok in sequence with the same splitter
+    // to ensure that the splitter can recover from a timeout
+    test_remote_chain_splitter_timeout_recover(test_ctx, &splitters.2)?;
+
+    info!("Finished remote chain splitter tests!");
+
+    Ok(())
+}
+
+fn start_relayer(test_ctx: &mut TestContext) {
+    let neutron = test_ctx.get_chain(NEUTRON_CHAIN_NAME);
+
+    reqwest::blocking::Client::default()
+        .post(&neutron.rb.api)
+        .json(&serde_json::json!({ "chain_id": NEUTRON_CHAIN_ID, "action": "start-relayer"}))
+        .send()
+        .unwrap();
+}
+
+fn stop_relayer(test_ctx: &mut TestContext) {
+    let neutron = test_ctx.get_chain(NEUTRON_CHAIN_NAME);
+
+    reqwest::blocking::Client::default()
+        .post(&neutron.rb.api)
+        .json(&serde_json::json!({ "chain_id": NEUTRON_CHAIN_ID, "action": "stop-relayer"}))
+        .send()
+        .unwrap();
+}
+
+fn upload_contracts(test_ctx: &mut TestContext) -> Result<(), LocalError> {
+    let mut uploader = test_ctx.build_tx_upload_contracts();
+
+    uploader
+        .send_with_local_cache(VALENCE_PATH, NEUTRON_CHAIN_NAME, LOCAL_CODE_ID_CACHE_PATH)
+        .unwrap();
+
+    uploader
+        .send_with_local_cache(ASTROPORT_PATH, NEUTRON_CHAIN_NAME, LOCAL_CODE_ID_CACHE_PATH)
+        .unwrap();
+
+    uploader.send().unwrap();
```

Comment:
I think this line is not needed and can be removed

Reply:
:+1: